bloom_nofos/users/auth/login_gov.py

```diff
@@ -0,0 +1,157 @@
+
+        # Get private key from settings
+        private_key = settings.LOGIN_GOV.get("PRIVATE_KEY")
+        if not private_key:
+            raise Exception(
+                "Private key not configured in settings.LOGIN_GOV['PRIVATE_KEY']"
+            )
+
+        try:
+            self.private_key = private_key.encode("utf-8")
+            load_pem_private_key(
+                self.private_key, password=None, backend=default_backend()
+            )
+        except Exception as e:
+            raise Exception(f"Error loading private key: {str(e)}")
+
+        self.redirect_uri = settings.LOGIN_GOV["REDIRECT_URI"].strip()
+        self.acr_values = settings.LOGIN_GOV["ACR_VALUES"]
+
+    def _get_login_gov_public_key(self, kid=None):
+        """Fetch Login.gov's public key configuration.
+
+        Args:
+            kid: Optional key ID to fetch a specific key
+        """
+        try:
+            certs_url = f"{self.token_url}/certs"
+            response = requests.get(certs_url)
+            response.raise_for_status()
+
+            keys = response.json()
+            if not keys or "keys" not in keys or not keys["keys"]:
+                raise ValueError("No keys found in Login.gov certs response")
+
+            if kid:
+                jwk = next((key for key in keys["keys"] if key.get("kid") == kid), None)
+                if not jwk:
+                    raise ValueError(f"No key found with ID: {kid}")
+            else:
+                jwk = next(
+                    (key for key in keys["keys"] if key.get("kty") == "RSA"), None
+                )
+                if not jwk:
+                    raise ValueError("No RSA key found in Login.gov certs")
+
+            return RSAAlgorithm.from_jwk(json.dumps(jwk))
+
+        except Exception as e:
+            raise ValueError(f"Error fetching Login.gov public key: {str(e)}")
+
+    def get_authorization_url(self, state=None):
+        """Generate the Login.gov authorization URL for the initial redirect."""
+        state = state or str(uuid.uuid4())
+        nonce = str(uuid.uuid4())
+
+        params = {
+            "client_id": self.client_id,
+            "response_type": "code",
+            "acr_values": self.acr_values,
+            "scope": "openid email",
+            "redirect_uri": self.redirect_uri,
+            "state": state,
+            "nonce": nonce,
+        }
+
+        auth_url = f"{self.oidc_url}/authorize?{urlencode(params)}"
+        return auth_url, state, nonce
+
+    def get_token(self, code):
+        """Exchange authorization code for tokens."""
+        if not self.private_key:
+            raise Exception(
+                "Private key not configured. Cannot create client assertion."
+            )
+
+        now = int(time.time())
+
+        try:
+            private_key = load_pem_private_key(
+                self.private_key, password=None, backend=default_backend()
+            )
```

Comment:
This is also called in `__init__` -- we could store the result of `load_pem_private_key` in there and save a couple lines of code in there, unless there's a reason not to.

Small thing though, not sure it matters very much.